tests/src/Unit/Plugin/migrate/process/PipelineTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_migration\Unit\Plugin\migrate\process;
+
+use Drupal\oe_migration\Entity\MigrationProcessPipelineInterface;
+use Drupal\migrate\MigrateException;
+use Drupal\Tests\migrate\Unit\process\MigrateProcessTestCase;
+use Drupal\oe_migration\Plugin\migrate\process\Pipeline;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\Entity\EntityTypeRepository;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+
+/**
+ * @coversDefaultClass \Drupal\oe_migration\Plugin\migrate\process\Pipeline
+ *
+ * @group oe_migration
+ */
+class PipelineTest extends MigrateProcessTestCase {
+
+  /**
+   * Name of the plugin tested.
+   *
+   * @var string
+   */
+  protected $pluginId = 'oe_migration_pipeline';
+
+  /**
+   * A MigrationProcessPipelineInterface mock.
+   *
+   * @var \Drupal\oe_migration\Entity\MigrationProcessPipelineInterface
+   */
+  protected $migrationProcessPipeline;
+
+  /**
+   * A valid process plugin id.
+   *
+   * @var string
+   */
+  protected $validId;
+
+  /**
+   * An invalid process plugin id.
+   *
+   * @var string
+   */
+  protected $invalidId;
+
+  /**
+   * Destination property, only to show errors.
+   *
+   * @var string
+   */
+  protected $destinationProperty = 'destination_property';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    $this->migrationProcessPipeline = $this->createMock(MigrationProcessPipelineInterface::class);
+    $this->validId = $this->randomMachineName();
+    $this->invalidId = $this->randomMachineName();
+  }
+
+  /**
+   * Test with a plugin id configured.
+   */
+  public function testInvalidConfiguration() {
+    $configuration = [];
+
+    try {
+      new Pipeline($configuration, $this->pluginId, []);
+    }
+    catch (\InvalidArgumentException | MigrateException $e) {
+      $this->assertEquals('The pipeline plugin requires a pipeline ID, none found.', $e->getMessage());
```

Comment:
As far as I see, this assertion will not cover a failure in the logic. For example, if the creation of the new Pipeline does NOT throw an exception, but it SHOULD, it's not asserted so it will pass while it shouldn't. Same issue for all assertions in this test that happen inside a catch.

Reply:
Indeed. I don't know why I didn't do it from the beginning. The related tests are already simplified too, using the the new injected class.

Reply:
Yes, you're right. I changed to enforce check that the exception is properly thrown. I think that right now the test is quite better.